### ISSUE TITLE
Non-functional requirements

### ISSUE THREAD

**Issue by cgendreau** [DINA-Web/documentation#2]
It would be very useful to identify the top [non-functional requirements](https://en.wikipedia.org/wiki/Non-functional_requirement) for the DINA system.
These requirements should guide general direction and influence architectural and design decisions. As a multi-modules, multi-institutions system, I would suggest the following list:

* Maintainability
* Testability
* Documentation (code and user documentation)
* Interoperability (e.g. JSON-API)
* Extensibility

It doesn't mean other non-functional requirements are not important but more that in case of conflict, the requirement from the list above should be prioritized. e.g. Maintainability vs Performance

**Comment by mikkohei13** [comment]
Agree, identifying these at high-level can guide the development and make sure if we agree on them. Currently some of there have been defined or suggested in the guidelines in more detailed way, but stating them on high-level can be more useful and give flexibility to find the a good implementation.
